src/app.jsx

```diff
@@ -30,35 +29,51 @@ import { NetworkList } from "./components/networks/networkList.jsx";
 import { VmDetailsPage } from './components/vm/vmDetailsPage.jsx';
 import { CreateVmAction } from "./components/create-vm-dialog/createVmDialog.jsx";
 import LibvirtSlate from "./components/libvirtSlate.jsx";
-import { isObjectEmpty, dummyVmsFilter, vmId } from "./helpers.js";
+import { dummyVmsFilter, vmId } from "./helpers.js";
 import { InlineNotification } from 'cockpit-components-inline-notification.jsx';
 import {
+    getApiData,
     usageStartPolling,
     usageStopPolling,
 } from "./libvirt-dbus.js";
+import { getOsInfoList, getLoggedInUser } from './libvirt-common.js';
 import { useObject, useEvent } from "hooks";
 import * as service from 'service.js';
+import store from './store.js';
 
-import {
-    updateLibvirtState,
-} from './actions/store-actions.js';
+import { updateLibvirtState } from './actions/store-actions.js';
 const _ = cockpit.gettext;
 
 superuser.reload_page_on_change();
 
-export const App = ({ store, name }) => {
+export const App = ({ name }) => {
     const libvirtService = useObject(() => service.proxy(name), null, []);
+    const prevLibvirtServiceRef = useRef();
+    const [loadingResources, setLoadingResources] = useState(false);
 
     useEvent(libvirtService, "changed", () => {
         store.dispatch(updateLibvirtState({
-            activeState: libvirtService.exists ? libvirtService.state : 'stopped',
+            activeState: libvirtService.state,
```

Comment:
If you need to touch this again, please squash this into the previous commit. (But no need to repush just for this)

Reply:
Done